src/app/modules/webhook/__tests__/webhook.service.spec.ts

```diff
@@ -0,0 +1,404 @@
+      )
+
+      expect(actual._unsafeUnwrapErr()).toEqual(expectedError)
+    })
+
+    it('should return updated submission with new webhook response if the record is successfully saved', async () => {
+      // Arrange
+      const mockWebhookResponse = {
+        _id: testEncryptedSubmission._id,
+        created: testEncryptedSubmission.created,
+        ...MOCK_WEBHOOK_SUCCESS_RESPONSE,
+        signature: testSignature,
+        webhookUrl: MOCK_WEBHOOK_URL,
+      } as IWebhookResponse
+
+      const expectedSubmission = new EncryptSubmissionModel({
+        ...testEncryptedSubmission,
+      })
+      expectedSubmission.webhookResponses = [mockWebhookResponse]
+
+      const addWebhookResponseSpy = jest
+        .spyOn(EncryptSubmissionModel, 'addWebhookResponse')
+        .mockResolvedValue(expectedSubmission)
+
+      // Act
+      const actual = await saveWebhookRecord(
+        testEncryptedSubmission._id,
+        mockWebhookResponse,
+      )
+
+      // Assert
+      expect(addWebhookResponseSpy).toHaveBeenCalledWith(
+        testEncryptedSubmission._id,
+        mockWebhookResponse,
+      )
+      expect(actual._unsafeUnwrap()).toEqual(expectedSubmission)
+    })
+  })
+
+  describe('sendWebhook', () => {
+    it('should return webhook url validation error if webhook url is not valid', async () => {
+      // Arrange
+      MockWebhookValidationModule.validateWebhookUrl.mockRejectedValueOnce(
+        new WebhookValidationError(DEFAULT_ERROR_MSG),
+      )
+
+      // Act
+      const actual = await sendWebhook(
+        testEncryptedSubmission,
+        MOCK_WEBHOOK_URL,
+      )
+
+      // Assert
+      const expectedError = new WebhookValidationError(DEFAULT_ERROR_MSG)
+
+      expect(actual._unsafeUnwrapErr()).toEqual(expectedError)
+    })
+
+    it('should return default webhook url validation error if webhook url is not valid and validate webhook url returns a non webhook url validation error', async () => {
+      // Arrange
+      MockWebhookValidationModule.validateWebhookUrl.mockRejectedValueOnce(
+        new Error(),
+      )
+
+      // Act
+      const actual = await sendWebhook(
+        testEncryptedSubmission,
+        MOCK_WEBHOOK_URL,
+      )
+
+      // Assert
+      const expectedError = new WebhookValidationError(
+        'Webhook URL is non-HTTPS or points to private IP',
+      )
+
+      expect(actual._unsafeUnwrapErr()).toEqual(expectedError)
+    })
+
+    it('should resolve with webhook failed with axios error message if axios post fails due to an axios error', async () => {
+      // Arrange
+      MockWebhookValidationModule.validateWebhookUrl.mockResolvedValueOnce()
+
+      const MOCK_AXIOS_ERROR: AxiosError = {
+        name: '',
+        message: AXIOS_ERROR_MSG,
+        config: {},
+        code: '',
+        response: MOCK_AXIOS_FAILURE_RESPONSE,
+        isAxiosError: true,
+        toJSON: () => jest.fn(),
+      }
+
+      MockAxios.post.mockRejectedValue(MOCK_AXIOS_ERROR)
```

Comment:
One of the mocks is called twice, so I thought it might be cleaner and more consistent to restore all mocks before each test is run to clear previous mocked values, then mock the values for each unit test.